src/test/java/fi/aalto/cs/apluscourses/ui/exercise/SubmissionDialogTest.java

```diff
@@ -43,31 +53,37 @@ public String getSubmissionNumberText() {
   }
 
   @NotNull
-  private SubmissionViewModel createMockViewModel(@NotNull String exerciseName,
-                                                  @NotNull List<Group> availableGroups,
-                                                  @NotNull SubmittableFile[] files,
-                                                  int numberOfSubmissions,
-                                                  int maxNumberOfSubmissions) {
-    SubmissionViewModel viewModel = mock(SubmissionViewModel.class);
-    doReturn(exerciseName).when(viewModel).getPresentableExerciseName();
-    doReturn(availableGroups).when(viewModel).getAvailableGroups();
-    doReturn(files).when(viewModel).getFiles();
-    doReturn("You are about to make submission 4 out of 10.")
-        .when(viewModel)
-        .getSubmissionCountText();
-    return viewModel;
+  private SubmissionViewModel createViewModel(@NotNull String exerciseName,
+                                              @NotNull List<Group> availableGroups,
+                                              @Nullable Group defaultGroup,
+                                              @NotNull List<SubmittableFile> files,
+                                              int numberOfSubmissions,
+                                              int maxNumberOfSubmissions) {
+    return new SubmissionViewModel(
+        new Exercise(1, exerciseName, "http://www.fi", 0, 0, maxNumberOfSubmissions),
```

Comment:
I haven't seen that URL for a long long time. IIRC, it used to be some quite popular Finnish search engine in the late 90s - early 2000s. 😃 Thanks for a nostalgic moment!

Reply:
I myself only very recently discovered that URL when I realized that there's nothing special about `www` in a URL :smile: